src/main/java/org/opensearch/geospatial/ip2geo/common/URLDenyListChecker.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.UnknownHostException;
+import java.util.List;
+
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.settings.ClusterSettings;
+
+import inet.ipaddr.IPAddressString;
+
+/**
+ * A class to check url against a deny-list
+ */
+public class URLDenyListChecker {
+    private final ClusterSettings clusterSettings;
+
+    public URLDenyListChecker(final ClusterSettings clusterSettings) {
+        this.clusterSettings = clusterSettings;
+    }
+
+    /**
+     * Convert String to URL after verifying the url is not on a deny-list
+     *
+     * @param url value to validate and convert to URL
+     * @return value in URL type
+     */
+    public URL toUrlIfNotInDenyList(final String url) {
+        try {
+            return toUrlIfNotInDenyList(url, clusterSettings.get(Ip2GeoSettings.DATASOURCE_ENDPOINT_DENYLIST));
+        } catch (UnknownHostException e) {
+            throw new IllegalArgumentException("host is unknown");
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("url is malformed");
```

Comment:
should we include the user input in the exception message?

Also, can we add a logger with info message with exception trace so that if required user can check in the logs what is proper exception.

Reply:
It is not recommended to return invalid user input in the message.
Let me add a log.

Reply:
`It is not recommended to return invalid user input in the message.`
sure..
lets add info log. and can we change this `url is malformed` to `URL provided in the datasource is malformed`